build-script.py

```diff
@@ -448,108 +450,155 @@ def install(build_dir, dylib_dir, swiftmodule_base_name, stdlib_rpath):
 
 
 # -----------------------------------------------------------------------------
+# Arugment Parsing
 
-def main():
-    parser = argparse.ArgumentParser(
-        formatter_class=argparse.RawDescriptionHelpFormatter,
-        description='''
+_DESCRIPTION = """
 Build and test script for SwiftSyntax.
 
 Build SwiftSyntax by generating all necessary files form the corresponding
 .swift.gyb files first. For this, SwiftSyntax needs to be check out alongside
 the main swift repo (http://github.com/apple/swift/) in the following structure
+
 - (containing directory)
   - swift
   - swift-syntax
+
 It is not necessary to build the compiler project.
 
 The build script can also drive the test suite included in the SwiftSyntax
 repo. This requires a custom build of the compiler project since it accesses
 test utilities that are not shipped as part of the toolchains. See the Testing
 section for arguments that need to be specified for this.
-''')
+"""
+
+
+_SWIFTMODULE_BASE_NAME_HELP = """
```

Comment:
Wouldn’t a local variable in `parse_args` be nicer than a global? Or was there a particular reason why you decided to go for the global variable?

Reply:
I tend to avoid local variable for this sort of string, even if it's only used in one place. Honestly I would say that this string is maybe too long for an argument help message.

Reply:
If you like it better, that’s fine. I don’t have too strong opinions on it.

The problem is that I haven’t found a shorter way to describe it. And putting it in the general help didn’t seem like the right solution either.

Reply:
Multi-line strings of this length are usually something I punt into the top-level scope, but under a "private" name. Makes the file more readable IMO.

Reply:
I guess it also avoid the issue of how you need to indent (or not indent) the string. Good with me. Let’s merge it.

Reply:
Sounds good!